### PR TITLE
working step0?

### PR DESCRIPTION
**Pull request by knguyen142**
Part of v02 migration. The strategy is to create a new v02 directory and go step by step and change only functions that make calls to Hail. 

This is the step0 change (except for validating, remapping/subset sampling, and vep). Only functionality that involves the load function for step0 are added.

The goal is to do multiple PRs step by step so the diffs are manageable. 

Most of this code is copied and pasted from their equivalent v01 file (we intentionally want to introduce as little change as possible for this port), so it makes sense to look at the equivalent v01 when reviewing.

Will send info on sanity checks/testing tomorrow.

### CODE REVIEW

File: hail_scripts/v02/load_dataset_to_es.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    # Store step0_output_vds as the cached version of the dataset in google buckets, and also set it as the global.sourceFilePath
+    # because
+    # 1) vep is the most time-consuming step (other than exporting to elasticsearch), so it makes sense to cache results
+    # 2) at this stage, all subsetting and remapping has already been applied, so the samples in the dataset are only the ones exported to elasticsearch
+    # 3) annotations may be updated / added more often than vep versions.
+    mt = mt.annotate_globals(sourceFilePath = args.step0_output_mt)
+    mt = mt.annotate_globals(genomeVersion = args.genome_version)
+    mt = mt.annotate_globals(sampleType = args.sample_type)
+    mt = mt.annotate_globals(datasetType = args.dataset_type)
```
Comment (nawatts) [inline hail_scripts/v02/load_dataset_to_es.py]: These could all be combined into one call to `annotate_globals`.

File: hail_scripts/v02/load_dataset_to_es.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    logger.info("Command args: \n" + " ".join(sys.argv[:1]) + ((" --index " + args.index) if "--index" not in sys.argv else ""))
+
+    logger.info("Parsed args: \n" + pformat(args.__dict__))
+
+    return args
+
+def compute_index_name(args):
+    """Returns elasticsearch index name computed based on command-line args"""
+
+    # generate the index name as:  <project>_<WGS_WES>_<family?>_<VARIANTS or SVs>_<YYYYMMDD>_<batch>
```
Comment (nawatts) [inline hail_scripts/v02/load_dataset_to_es.py]: The format in this comment doesn't look accurate.

File: hail_scripts/v02/load_dataset_to_es.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+from hail_scripts.v01.utils.fam_file_utils import MAX_SAMPLES_PER_INDEX, compute_sample_groups_from_fam_file
+from hail_scripts.v01.utils.vds_schema_string_utils import convert_vds_schema_string_to_annotate_variants_expr
+from hail_scripts.v01.utils.add_1kg_phase3 import add_1kg_phase3_to_vds
+from hail_scripts.v01.utils.add_cadd import add_cadd_to_vds
+from hail_scripts.v01.utils.add_dbnsfp import add_dbnsfp_to_vds
+from hail_scripts.v01.utils.add_clinvar import add_clinvar_to_vds
+from hail_scripts.v01.utils.add_exac import add_exac_to_vds
+from hail_scripts.v01.utils.add_gnomad import add_gnomad_to_vds
+from hail_scripts.v01.utils.add_topmed import add_topmed_to_vds
+from hail_scripts.v01.utils.add_mpc import add_mpc_to_vds
```
[the remark this answers is not in the document]
Comment (knguyen142) [reply hail_scripts/v02/load_dataset_to_es.py]: Essentially, yes.

So the context is we decided to just change the minimal possible (only the hail diff), so copied and pasted everything from v01 to this. But instead of doing a PR with the whole file (since we can't see the PR diff changes in a new file),I decided to take everything out of that are the later steps so the PR is more manageable.

I left it here so as I add the other steps I won't have to worry about imports, and then when it's all done I'll remove the unused ones.

tldr; a middle ground between a way of sanity for me so I don't have to look at all the code at once, and a way for PR reviewers to not have to look at all of the non-relevant v01 code at once.

[the remark this answers is not in the document]
Comment (knguyen142) [reply hail_scripts/v02/load_dataset_to_es.py]: Now that I'm thinking about it, it might make sense for me to get a copied and pasted version of v01 to v02 merged to master, and all subsequent diffs on v02 will only see v02 changes against a v01 copy.

**Comment by knguyen142** [comment]
New strategy, will get a base copy of v02 (copied from v01) merged to master, and then do a PR again so we can just view the diffs I made in an easy way.
@nawatts will address your PR comments in new PR.

File: hail_scripts/v02/utils/vds_utils.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import hail as hl
```
Comment (knguyen142) [inline hail_scripts/v02/utils/vds_utils.py]: And why did I name this file vds_utils? Haha.
